app/controllers/declaration/ItemsSummaryController.scala

```diff
@@ -43,35 +43,26 @@ class ItemsSummaryController @Inject()(
 
   def addItem(): Action[AnyContent] = (authenticate andThen journeyType).async { implicit request =>
     val newItem = ExportItem(id = exportItemIdGeneratorService.generateItemId())
-    updateCache(journeySessionId, newItem)
+    exportsCacheService
+      .update(
+        journeySessionId,
+        request.cacheModel
+          .copy(items = request.cacheModel.items + newItem.copy(sequenceId = request.cacheModel.items.size + 1))
+      )
       .map(_ => Redirect(controllers.declaration.routes.ProcedureCodesController.displayPage(newItem.id)))
   }
 
   def removeItem(itemId: String): Action[AnyContent] = (authenticate andThen journeyType).async { implicit request =>
-    exportsCacheService.get(journeySessionId) flatMap {
-      case Some(model) =>
-        model.items.find(_.id == itemId) match {
-          case Some(itemToDelete) =>
-            val updatedItems = model.copy(items = model.items - itemToDelete).items.zipWithIndex.map {
-              case (item, index) => item.copy(sequenceId = index + 1)
-            }
-            exportsCacheService.update(journeySessionId, model.copy(items = updatedItems)).map { _ =>
-              Redirect(routes.ItemsSummaryController.displayPage())
-            }
-          case _ =>
-            Future.successful(Redirect(routes.ItemsSummaryController.displayPage()))
+    request.cacheModel.items.find(_.id == itemId) match {
```

Comment:
isnt there a helper method on `ExportsDeclaration` for finding items by id?

Reply:
Thank you! Changed :)